### PR TITLE
✨ introduce proper logging

### LINKED ISSUE #68: Introduce proper logging
Currently there are print statements spread out in the client. These need to be converted into proper a proper logging system

### PR DESCRIPTION
**Pull request by bisgaard-itis**
## What do these changes do?
- Introduce a proper logger. After running a study from Taylor I realize how important it is that we have  a proper logger in place.

## Related issue/s
- resolves #68

### CODE REVIEW

**Review by pcrespov — APPROVED** [review]
This looks great!
left some suggestions

File: clients/python/client/osparc/_files_api.py [hunk 1]
```diff
@@ -113,41 +117,44 @@ async def upload_file_async(
...
-                            chunck=chunck,
-                            chunck_size=size,
-                            upload_link=url,
-                            index=index,
+                with logging_redirect_tqdm():
+                    _logger.info("uploading chunks...")
+                    async for chunck, size in tqdm(
+                        file_chunk_generator(file, chunk_size),
+                        total=n_urls,
+                        disable=(not _logger.isEnabledFor(logging.INFO)),
```
[the remark this answers is not in the document]
Comment (bisgaard-itis) [reply clients/python/client/osparc/_files_api.py:128]: I checked that actually the comparison of the `_logger.level` doesn't work in this case. For example if I set `logging.basicConfig(level=logging.INFO)` then `_logger.level != logging.INFO`. On the other hand I checked that `_logger.isEnabledFor(logging.INFO)` returns `True` if and only if the `_logger` will print `logging.INFO`-level statements.

[the remark this answers is not in the document]
Comment (bisgaard-itis) [reply clients/python/client/osparc/_files_api.py:128]: This system is not perfect, but I researched quite a bit which progress bar to use in python which both supports asyncio and can be enabled/disabled using log-levels and this seemed to be the simplest approach.

File: clients/python/client/osparc/_files_api.py [hunk 2]
```diff
@@ -113,41 +117,44 @@ async def upload_file_async(
...
+                        configuration=self.api_client.configuration,
+                        request_type="post",
+                        url=links.abort_upload,
+                        base_url=self.api_client.configuration.host,
+                        follow_redirects=True,
+                        auth=self._auth,
+                        timeout=timeout_seconds,
+                    ) as session:
+                        _logger.info(
+                            "completing upload (this might take a couple of minutes)..."
```
[the remark this answers is not in the document]
Comment (bisgaard-itis) [reply clients/python/client/osparc/_files_api.py:151]: Done

File: clients/python/client/osparc/_files_api.py [hunk 3]
```diff
@@ -113,41 +117,44 @@ async def upload_file_async(
...
-                    index, url = next(url_iter)
-                    uploaded_parts.append(
-                        await self._upload_chunck(
-                            http_client=session,
-                            chunck=chunck,
-                            chunck_size=size,
-                            upload_link=url,
-                            index=index,
+                with logging_redirect_tqdm():
+                    _logger.info("uploading chunks...")
```
[the remark this answers is not in the document]
Comment (bisgaard-itis) [reply clients/python/client/osparc/_files_api.py:124]: Done

**Pull request merged**